diff --git a/src/Remapping/TempestRemapper.cpp b/src/Remapping/TempestRemapper.cpp
index e168742c4..e55e6aaa7 100644
--- a/src/Remapping/TempestRemapper.cpp
+++ b/src/Remapping/TempestRemapper.cpp
@@ -1248,42 +1248,45 @@ ErrorCode TempestRemapper::ComputeOverlapMesh( bool kdtree_search, bool use_temp
             // remove from the set !
             if( !point_cloud_target )
             {
-                Range covEnts;
+                std::vector<EntityHandle> covEnts;
                 rval = m_interface->get_entities_by_dimension( m_covering_source_set, 2, covEnts );MB_CHK_ERR( rval );
                 Tag gidtag = m_interface->globalId_tag();
 
                 std::map< int, int > loc_gid_to_lid_covsrc;
                 std::vector< int > gids( covEnts.size(), -1 );
-                rval = m_interface->tag_get_data( gidtag, covEnts, &gids[0] );MB_CHK_ERR( rval );
+                rval = m_interface->tag_get_data( gidtag, covEnts.data(), covEnts.size(), &gids[0] );MB_CHK_ERR( rval );
                 for( unsigned ie = 0; ie < gids.size(); ++ie )
                 {
                     loc_gid_to_lid_covsrc[gids[ie]] = ie;
                 }
 
-                Range intxCov;
-                Range intxCells;
-                Tag srcParentTag;
-                rval = m_interface->tag_get_handle( "SourceParent", srcParentTag );MB_CHK_ERR( rval );
-                rval = m_interface->get_entities_by_dimension( m_overlap_set, 2, intxCells );MB_CHK_ERR( rval );
-                for( Range::iterator it = intxCells.begin(); it != intxCells.end(); it++ )
-                {
-                    EntityHandle intxCell = *it;
-                    int blueParent        = -1;
-                    rval                  = m_interface->tag_get_data( srcParentTag, &intxCell, 1, &blueParent );MB_CHK_ERR( rval );
-                    // if (is_root) std::cout << "Found intersecting element: " << blueParent << ",
-                    // " << gid_to_lid_covsrc[blueParent] << "\n";
-                    assert( blueParent >= 0 );
-                    intxCov.insert( covEnts[loc_gid_to_lid_covsrc[blueParent]] );
-                }
-
                 if( size > 1 )
                 {
-                    Range notNeededCovCells = moab::subtract( covEnts, intxCov );
+//                    std::vector<EntityHandle> intxCov;
+                    std::vector<EntityHandle> notNeededCovCells;
+                    Range intxCells;
+                    Tag srcParentTag;
+                    std::vector<int> blueParent(intxCells.size());
+                    rval = m_interface->tag_get_handle( "SourceParent", srcParentTag );MB_CHK_ERR( rval );
+                    rval = m_interface->tag_get_data( srcParentTag, intxCells, blueParent.data() );MB_CHK_ERR( rval );
+                    rval = m_interface->get_entities_by_dimension( m_overlap_set, 2, intxCells );MB_CHK_ERR( rval );
+                    for( size_t it = 0; it < intxCells.size(); it++ )
+                    {
+                        assert( blueParent[it] >= 0 );
+                        EntityHandle inactiveCell = loc_gid_to_lid_covsrc[blueParent[it]];
+                        covEnts.erase(std::remove(covEnts.begin(), covEnts.end(), inactiveCell), covEnts.end());
+                        notNeededCovCells.emplace_back(inactiveCell);
+//                        intxCov.push_back( covEnts[loc_gid_to_lid_covsrc[blueParent[it]]] );
+                    }
+                    
+//                    Range notNeededCovCells = covEnts.erase(intxCov.begin(), intxCov.end());
+                    
+//                    Range notNeededCovCells = moab::subtract( covEnts, intxCov );
                     // remove now from coverage set the cells that are not needed
-                    rval = m_interface->remove_entities( m_covering_source_set, notNeededCovCells );MB_CHK_ERR( rval );
-                    covEnts = moab::subtract( covEnts, notNeededCovCells );
+                    rval = m_interface->remove_entities( m_covering_source_set, &notNeededCovCells[0], notNeededCovCells.size() );MB_CHK_ERR( rval );
+//                    covEnts = moab::subtract( covEnts, notNeededCovCells );
 #ifdef VERBOSE
-                    std::cout << " total participating elements in the covering set: " << intxCov.size() << "\n";
+                    std::cout << " total participating elements in the covering set: " << covEnts.size() << "\n";
                     std::cout << " remove from coverage set elements that are not intersected: "
                               << notNeededCovCells.size() << "\n";
 #endif
